c2/c2-protocol/c2-protocol-component-api/src/main/java/org/apache/nifi/c2/protocol/component/api/RuntimeManifest.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.c2.protocol.component.api;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+@ApiModel
+public class RuntimeManifest implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private String identifier;
+    private String runtimeType;
+    private String version;
+    private BuildInfo buildInfo;
+    private List<Bundle> bundles;
+    private SchedulingDefaults schedulingDefaults;
+
+    @ApiModelProperty("A unique identifier for the manifest")
+    public String getIdentifier() {
+        return identifier;
+    }
+
+    public void setIdentifier(String identifier) {
+        this.identifier = identifier;
+    }
+
+    @ApiModelProperty(
+        value = "The type of the runtime binary, e.g., 'minifi-java' or 'minifi-cpp'",
+        notes = "Usually set when the runtime is built.")
+    public String getRuntimeType() {
+        return runtimeType;
+    }
+
+    public void setRuntimeType(String runtimeType) {
+        this.runtimeType = runtimeType;
+    }
```

Comment:
`runtimeType` is indeed more accurate and consistent with renaming AgentManifest to RuntimeManifest. Unfortunately, in order to maintain compatibility with nifi-minifi-cpp, this field (or at least the getters/setters that will be used for serialization/deserialization) should stay `agentType` for now.

Reply:
I think a similar case is in `AgentInfo` where it had a field of `AgentManifest`, the field name needs to remain `agentManifest` for compatibility with minifi-cpp. In this case it looks like the getter/setter is still AgentManifest, only the field is runtimeManifest, so I suppose it depends if the unmarshalling is using fields or methods.